src/mesh.cpp

```diff
@@ -181,6 +181,87 @@ void readData(Mesh &mesh, const std::string &filename)
         break;
       }
     }
+#ifdef ASTE_NN_GRADIENT_MAPPING
+    else if (datatype == aste::datatype::GRADIENT) {
+      const int gradDim = data.gradDimension; // Number of components of gradient
```

Comment:
Is this the number of components per direction or the total number of gradient components? Can you state it somewhere?

Reply:
This is basically the dimension of the problem. For vectors, we know the dimension of the problem but for scalars, we need this information